Drop-In/src/main/java/com/braintreepayments/api/DropInEvent.java

```diff
@@ -0,0 +1,107 @@
+package com.braintreepayments.api;
+
+import android.os.Bundle;
+import android.os.Parcel;
+import android.os.Parcelable;
+
+class DropInEvent {
+
+    static final String REQUEST_KEY = "DROP_IN_EVENT_REQUEST_KEY";
+
+    private static final String TYPE_KEY = "DROP_IN_EVENT_TYPE";
+
+    static DropInEvent createSendAnalyticsEvent(String eventName) {
+        DropInEvent event = new DropInEvent(DropInEventType.SEND_ANALYTICS);
+        event.putString(DropInEventProperty.ANALYTICS_EVENT_NAME, eventName);
+        return event;
+    }
+
+    static DropInEvent createAddCardSubmitEvent(String cardNumber) {
+        DropInEvent event = new DropInEvent(DropInEventType.ADD_CARD_SUBMIT);
+        event.putString(DropInEventProperty.CARD_NUMBER, cardNumber);
+        return event;
+    }
+
+    static DropInEvent createCardDetailsSubmitEvent(Card card) {
+        DropInEvent event = new DropInEvent(DropInEventType.CARD_DETAILS_SUBMIT);
+        event.putParcelable(DropInEventProperty.CARD, card);
+        return event;
+    }
+
+    static DropInEvent createEditCardEvent(String cardNumber) {
+        DropInEvent event = new DropInEvent(DropInEventType.EDIT_CARD_NUMBER);
+        event.putString(DropInEventProperty.CARD_NUMBER, cardNumber);
+        return event;
+    }
+
+    static DropInEvent createDeleteVaultedPaymentMethodNonceEvent(PaymentMethodNonce paymentMethodNonceToDelete) {
+        DropInEvent event = new DropInEvent(DropInEventType.DELETE_VAULTED_PAYMENT_METHOD);
+        event.putParcelable(DropInEventProperty.VAULTED_PAYMENT_METHOD, paymentMethodNonceToDelete);
+        return event;
+    }
+
+    static DropInEvent createVaultedPaymentMethodSelectedEvent(PaymentMethodNonce paymentMethodNonce) {
+        DropInEvent event = new DropInEvent(DropInEventType.VAULTED_PAYMENT_METHOD_SELECTED);
+        event.putParcelable(DropInEventProperty.VAULTED_PAYMENT_METHOD, paymentMethodNonce);
+        return event;
+    }
+
+    static DropInEvent createSupportedPaymentMethodSelectedEvent(DropInPaymentMethodType type) {
```

Comment:
Nitpick - do we need to specify that the payment method is `supported`. Don't we only show supported payment methods?

Reply:
In my opinion it's nice to differentiate between a vaulted payment method and a new (supported) payment method, but maybe there is a better name for those?

Reply:
Yeah I agree in general. I'd be down to make these more concise if possible. The factory methods do make it easier when constructing an event though. It allows us to keep event creation logic in one place for the most part.